### DISCUSSION TITLE
canceling print job on E3V2 display.

### DISCUSSION THREAD

**Discussion by cruzer619** [Jyers/Marlin#366] [Q&A] [answered]
Just wanted to see if maybe I have a setting not set...  but when I cancel a print job on the E3V2 Display.  And then print again.. only "printing.." shows on the display when I do another print.  Instead of displaying normal display settings.  When I turn off/on printer.. print jobs are displayed again on the display.
![20210401_194812](https://user-images.githubusercontent.com/26581299/113375142-a9646700-9323-11eb-9362-1c01caeb5256.jpg)

**Comment by Jyers** [comment]
Are you printing from SD or OctoPrint?

**Comment by cruzer619** [comment]
from OctoPrint.

**Reply by Jyers** [reply]
Make sure your settings match the wiki page https://github.com/Jyers/Marlin/wiki/OctoPrint-Settings

**Reply by cruzer619** [reply]
sure thing, will double check.. but I know I copied the wiki line for line... will let you know.

**Reply by cruzer619** [reply]
These are my settings.. they match..  with exception of my custom g-codes.. but i'll remove'em on a test print and see if it makes a difference.  :

Before print job starts 		:  	M75 ; Start Print Job on Display
									M117 <F>{{ event.name }} ;Send Filename to Display
									M412 S0 ;Disable filament sensor

After print job completes 		: 	M77 ;Stop Print Job on Display

After print job is cancelled	:	M77 ;Pause Print Job on Display
									G1 X5 Y5 F3000 ;Wipe out
									G1 Z10 ;Raise Z more
									M106 S0 ;disable fan
									G1 X0 Y180 ;Present print
									M84 ; disable motors

After print job is paused		:	M76 ;Pause Print Job on Display

Before print job is resumed		:	M75 ;Start Print Job on Display

**Reply by Jyers** [reply]
That all looks correct, let me know if the issue persists and if there are some steps I can use to try to recreate it.

**Reply by cruzer619** [reply]
ok not sure what I did or what happened.. but I was still getting the same issue when removing my custom g-code.  so I removed all g-code.  Saved it.. restarted my Pi.  Then entered in default g-code of your wiki.  And now its working as expected.. even have my custom gcode back and still working as it should.. Now I have a new problem... when I cancel a job turns off hotend and bed, going to assume M77 turns off bed and hotend.  If I remove M77 from gcode area, works like before, keeping hotend and bed on which is what I want ( when I cancel from Octoprint ).  So am I going to have to use M104 and M140 then?

**Comment by cruzer619** [comment, marked as answer]
Ok got it figured it out.  Since M77 does turn off hotend and bed..  your going to have to add gcode to get back Octoprint default behavior.  Also really quick, when playing around with gcode.  I got the error again on display "printing..." going to assume I had some space or hidden text in the text box.  Making the gcode not to function right.  So my original error going to put it on a typo.

Well this is my gcode now with bed and hotend still staying on.  *might be a good idea to put that on the wiki*

M77 ;Cancel Print Job on Display
G1 X5 Y5 F3000 ;Wipe out
G1 Z10 ;Raise Z more
M106 S0 ;disable fan
G1 X0 Y180 ;Present print
M84 ; disable motors
M140 S{{last_temperature['b']['target']}} ; Bed on Last Temp
M104 S{{last_temperature[0]['target']}} ; Hotend on Last Temp

**Reply by Jyers** [reply]
Glad to here you solved the issue! I don't think I'll add those setting to the wiki as I believe most people will want octoprint to shut off the heaters after a cancel, as that is how marlin handles it by default. This discussion will be a good reference for anyone looking for a similar setup though!

**Reply by cruzer619** [reply]
understood.. just saying default behavior of Octoprint would leave them on, on a cancel.
